jib-core/src/main/java/com/google/cloud/tools/jib/frontend/ParameterValidator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.frontend;
+
+import com.google.cloud.tools.jib.builder.BuildLogger;
+import java.util.List;
+
+/** Used for validating configuration parameters. */
+public class ParameterValidator {
+
+  /**
+   * Checks a list for null or empty elements. Warns the user if an empty string is found, and
+   * throws an error if a null element is found.
+   *
+   * @param list the list to check
+   * @param parameterName the name of the configuration parameter being checked
+   * @param logger used for displaying messages
+   */
+  public static void checkListForNullOrEmpty(
+      List<String> list, String parameterName, BuildLogger logger) {
+    boolean foundNull = false;
+    boolean foundWarning = false;
+    for (String element : list) {
+      if (element == null) {
+        foundNull = true;
+        break;
+      } else if (element.equals("")) {
+        foundWarning = true;
+      }
+    }
+
+    if (foundNull) {
+      throw new IllegalStateException(
```

Comment:
This can just go on line 39. `IllegalStateException` is probably not the exception you want to use. This exception is meant to be used when there is an illegal state that our program reached, which means that there was some programming error. You would want some checked exception here. Maybe throw a GradleException/MavenExecutionException?

Reply:
Needs to be something else since this is in jib-core. RuntimeException?

Reply:
You could parameterize the exception to throw like

`<T extends Throwable> method(Supplier<T> throwableSupplier) throws T;`